diff --git a/BUILD.gn b/BUILD.gn
index 0a1f12b5de..ce5f7ccf73 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -4270,6 +4270,34 @@ v8_executable("v8_hello_world") {
   ]
 }
 
+v8_executable("erlang_v8") {
+  sources = [
+    "../../c_src/erlang_v8.cc",
+    "../../c_src/erlang_v8.h",
+    "../../c_src/debug.h",
+    "../../c_src/packet.h",
+    "../../c_src/report.cc",
+    "../../c_src/report.h",
+    "../../c_src/vm.cc",
+    "../../c_src/vm.h",
+  ]
+
+  configs = [
+    # Note: don't use :internal_config here because this target will get
+    # the :external_config applied to it by virtue of depending on :v8, and
+    # you can't have both applied to the same target.
+    ":internal_config_base",
+  ]
+
+  deps = [
+    ":v8",
+    ":v8_libbase",
+    ":v8_libplatform",
+    "//build/win:default_exe_manifest",
+  ]
+}
+
+
 v8_executable("v8_sample_process") {
   sources = [
     "samples/process.cc",
